scd/token.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of trust|me
+ * Copyright(c) 2013 - 2020 Fraunhofer AISEC
+ * Fraunhofer-Gesellschaft zur Förderung der angewandten Forschung e.V.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2 (GPL 2), as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GPL 2 license for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, see <http://www.gnu.org/licenses/>
+ *
+ * The full GNU General Public License is included in this distribution in
+ * the file called "COPYING".
+ *
+ * Contact Information:
+ * Fraunhofer AISEC <trustme@aisec.fraunhofer.de>
+ */
+
+#ifndef TOKEN_H
+#define TOKEN_H
+
+#include "softtoken.h"
+#include "usbtoken.h"
+
+#include "common/uuid.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#define STOKEN_DEFAULT_PASS "trustme"
+
+/**
+ *  Generic token type
+ */
+typedef struct scd_token scd_token_t;
+
+/**
+ * Internal data if the generic scd token.
+ */
+typedef struct scd_token_data scd_token_data_t;
+
+/**
+ * Choice of supported token types.
+ * Must be kept in sync with scd.proto
+ */
+typedef enum scd_tokentype { NONE, SOFT, USB } scd_tokentype_t;
+
+/**
+ * Data used by the constructor scd_token_new
+ */
+typedef struct token_constr_data {
+	scd_tokentype_t type;
+	char *uuid;
+
+	union {
+		const char *softtoken_dir;
+		const char *usbtoken_serial;
+	} str;
```

Comment:
confusing member name str

Reply:
it is confusing because it is sematically overloaded. the string expresses either the fs diecoty where the softtoken should be stored or the iSerial of the usbtoken reader. any suggestions for a better name?

Reply:
i would use extra_arg, extra_param, ...
Any name that indicates the purpose of that member